### PR TITLE
[USBSTOR] - Fixing USBSTOR_GetMaxLUN()

### PR DESCRIPTION
**Pull request by vgalnt**
"USB Mass Storage Class. Bulk-Only Transport. Revision 1.0".
  3.2  Get Max LUN (class-specific request):
     Devices that do not support multiple LUNs may STALL this command.

A multiple LUNs have for example card readers (with USB interface).

### CODE REVIEW

File: reactos/drivers/usb/usbstor/misc.c [hunk 1]
```diff
@@ -369,6 +369,8 @@ USBSTOR_GetMaxLUN(
         return STATUS_INSUFFICIENT_RESOURCES;
     }
 
+    *Buffer = 0;
```
Comment (ThFabba) [inline reactos/drivers/usb/usbstor/misc.c]: AllocateItem zeroes the memory so this is not necessary.

File: reactos/drivers/usb/usbstor/misc.c [hunk 2]
```diff
@@ -376,19 +378,34 @@ USBSTOR_GetMaxLUN(
...
-        //
-        // invalid response documented in usb mass storage specification
-        //
-        Status = STATUS_DEVICE_DATA_ERROR;
+        if (*Buffer > 0xF)
+        {
+            //
+            // invalid response documented in usb mass storage specification
+            //
+            Status = STATUS_DEVICE_DATA_ERROR;
```
Comment (ThFabba) [inline reactos/drivers/usb/usbstor/misc.c]: Should DeviceExtension->MaxLUN be set to something in this case?

Comment (vgalnt) [inline reactos/drivers/usb/usbstor/misc.c]: No. In this case return STATUS_DEVICE_DATA_ERROR :
```
USBSTOR_FdoHandleStartDevice()
{
    Status = USBSTOR_GetMaxLUN(DeviceExtension->LowerDeviceObject, DeviceExtension);
    if (!NT_SUCCESS(Status))
    {
        return Status;
    }
}
```

File: reactos/drivers/usb/usbstor/misc.c (hunk 1 above)
Comment (vgalnt) [inline reactos/drivers/usb/usbstor/misc.c]: Yes.

File: reactos/drivers/usb/usbstor/misc.c (hunk 2 above)
Comment (ThFabba) [inline reactos/drivers/usb/usbstor/misc.c]: Yup that makes sense, thanks.

**Comment by AmineKhaldi** [comment]
Work committed in r71128. https://jira.reactos.org/browse/CORE-10515
Thank you Vadim, keep up the great work!

**Pull request closed without merging**